### ISSUE TITLE
Swift Package Manager integration already broken (4.0.0), though it's not a pain

### ISSUE THREAD

**Issue by dreampiggy** [Instagram/IGListKit#1406]
## New issue checklist

- [x] I have reviewed the [`README`](https://github.com/Instagram/IGListKit/blob/master/README.md) and [documentation](http://instagram.github.io/IGListKit)
- [x] I have searched [existing issues](https://github.com/Instagram/IGListKit/issues) and this is not a duplicate
- [x] I have attempted to reproduce the issue and include an example project.

### General information

- `IGListKit` version: 4.0.0
- iOS version(s): Any
- CocoaPods/Carthage version: None
- Xcode version: Xcode 11.2.1
- Devices/Simulators affected: Any
- Reproducible in the demo project? (Yes/No): Yes
- Related issues:

### Debug information

```bash
# Please include debug logs using the following lldb command:
po [IGListDebugger dump]
```

Just two screenshots, say the issue currently.

![image](https://user-images.githubusercontent.com/6919743/70725758-eb852600-1d37-11ea-8713-e790bb26acf5.png)

![image](https://user-images.githubusercontent.com/6919743/70725765-ef18ad00-1d37-11ea-87cb-5786b9fa4101.png)

**Comment by danqing** [comment]
SPM isn't supported (https://github.com/Instagram/IGListKit/issues/1368).

Any idea when it can be added? @lorixx :)

**Comment by dreampiggy** [comment]
@danqing For SwiftPM on Objective-C, it's a pain because SwiftPM design assume (using the raw C style for `include` public headers), however:

Objective-C developer used to use `Object.h/Object.m` in the same directory, not a standalone `include` directory for public headers. Tha's a suck.

If you want to supports SwiftPM, you can checkout what [SDWebImage](https://github.com/SDWebImage/SDWebImage/blob/master/Package.swift) do, which have also two Target (one `SDWebImage` and another `SDWebImageMapKit`). Which use the custom header search path to achieve this, and also have to hide the `Private Headers` (I previouslly think facebook guys already solve this problem, but actually not)

**Comment by Bruce-pac** [comment]
@dreampiggy I note that SDWebImage use `#import ".h"` to import its file. And I try that way on IGListKit and write the Package.swift, it works. So Is that what you mean above you said？

**Comment by dreampiggy** [comment]
@Bruce-pac Use both 

+ Objective-C

```objective-c
@import IGListKit;
```

+ Swift

```swift
import IGListKit
```

cause the compile error showed in the first screenshot. You can have a check again at 4.0.0 version via SwiftPM.

**Comment by Bruce-pac** [comment]
@dreampiggy yeah, a big reason of that is that the Package.swift manifest file is not right. I'm trying rewrite Package.swift file.
I note that SDWebImage use `#import ".h"` to import its file in its other file. such as this:
`#import "SDAnimatedImage.h"
#import "NSImage+Compatibility.h"
#import "SDImageCoder.h"
#import "SDImageCodersManager.h"
#import "SDImageFrame.h"
#import "UIImage+MemoryCacheCost.h"
#import "SDImageAssetManager.h"`
However, IGListKit use `#import <A/B.h>` to import its file in its other file.
The modified file like:
`// swift-tools-version:5.1
import PackageDescription

let package = Package(name: "IGListKit",
                      platforms: [
                          .macOS(.v10_11),
                          .iOS(.v9),
                          .tvOS(.v9)
                      ],
                     products: [
//                          .library(name: "IGListKit", targets: ["IGListKit"]),
                          .library(name: "IGListDiffKit", targets: ["IGListDiffKit"])
                      ],
                     dependencies: [],
                     targets: [.target(name: "IGListDiffKit",
                               dependencies: [],
                               path: nil,
                               sources: nil,
                               publicHeadersPath: "Source/IGListDiffKit/Public",
                               cSettings: [
                               .headerSearchPath("Public"),
                               .headerSearchPath("Internal"),
                             ],
                               linkerSettings: [
                             .linkedLibrary("libc++"),
                             ]
                     ),
//                          .target(name: "IGListKit",
//                                  dependencies: ["IGListDiffKit"],
//                                  path: ".",
//                                  sources: ["Source/IGListKit"],
//                                  publicHeadersPath: "include/IGListKit",
//                                  cSettings: [
//                                    .headerSearchPath("Source/IGListKit"),
//                                    .headerSearchPath("Source/IGListKit/Internal"),
//                                    .headerSearchPath(".")
//                                ],
//                                  linkerSettings: [
//                                  .linkedLibrary("libc++"),
//                                  ]
//                                )
    ],
                     cxxLanguageStandard: .cxx11)
`
If I modify the import style to `#import ".h"` , IGListDiffKit can compile success, but the original import style `#import <A/B.h>` cause the compile error showed in the first screenshot.

So I guess this issue maybe has two solutions, one is like SDWebImage, other is like [CocoaLumberjack](https://github.com/CocoaLumberjack/CocoaLumberjack).

**Comment by dreampiggy** [comment]
If you use `#import <A/B.h>`. You must have public header search path, pointer to `A`, A is a directory.

For example, move the source code's header, as this strcture

```
- Source <---- Public Header Directory
-- IGListKit
--- IGListAdapter.h
```

So the SwiftPM can search this with `<IGListKit/IGListAdapter.h>`

**Comment by claudiogomezGL** [comment]
Do you have an ETA on when this is gonna be released? @Bruce-pac @dreampiggy

**Comment by jafara** [comment]
spm sucks, i stick with cocoapods

**Comment by lorixx** [comment]
OMG I had exact issue that @Bruce-pac experienced before: https://github.com/Instagram/IGListKit/issues/1368#issuecomment-696529398 this is what I am trying, maybe we cannot go around the Apple's setting to require "XXX.h" format instead of <Library/XXX>h> format here.

@dreampiggy interesting, let me test it out and see

**Comment by dreampiggy** [comment]
I'm not the maintainer of IGListKit. I'm the maintainer of SDWebImge.

You can try to see how we use symbol link the header to solve the import syntax of `#import <SDWebImage/SDWebImage.h>` across all these Package Managers:

+ CocoaPods (including static library/framework, use modular headers, 3 types)
+ Carthage (prebuilt binary framework)
+ SwiftPM

https://github.com/SDWebImage/SDWebImage/pull/2987

**Comment by 3a4oT** [comment]
Some work in progress #1465

**Comment by BrentMifsud** [comment]
any updates on this?

**Comment by TimOliver** [comment]
Hey folks! SPM support is officially added now. Sorry for the long wait. Please let me know if you have any questions about it! Thanks! :)
